File: FbxOpenCpp/ClusterTest.cpp

#include "Tests.h"

using namespace std;

void Cluster_SetLink_SetsLink()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxCluster* cluster = FbxCluster::Create(manager, "");
    FbxNode* node = FbxNode::Create(manager, "");

    // require:
    AssertEqual(0, cluster->GetSrcObjectCount());
    AssertEqual(0, cluster->GetDstObjectCount());
    AssertNull(cluster->GetLink());

    AssertEqual(0, node->GetSrcObjectCount());
    AssertEqual(0, node->GetDstObjectCount());

    // when:
    cluster->SetLink(node);

    // then:
    AssertEqual(1, cluster->GetSrcObjectCount());
    AssertEqual(node, cluster->GetSrcObject(0));
    AssertEqual(0, cluster->GetDstObjectCount());
    AssertEqual(node, cluster->GetLink());

    AssertEqual(0, node->GetSrcObjectCount());
    AssertEqual(1, node->GetDstObjectCount());
    AssertEqual(cluster, node->GetDstObject(0));
}

void FbxCluster_Create_HasNamespacePrefix()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxCluster* obj = FbxCluster::Create(manager, "asdf");

    // then:
    AssertEqual("SubDeformer::", obj->GetNameSpacePrefix());
}

void ClusterTest::RegisterTestCases()
{
    AddTestCase(Cluster_SetLink_SetsLink);
    AddTestCase(FbxCluster_Create_HasNamespacePrefix);
}

File: FbxSharpTests/ClusterTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class ClusterTest : TestBase
    {
        [Test]
        public void Cluster_SetLink_SetsLink()
        {
            // given:
            var cluster = new FbxCluster("");
            var node = new FbxNode("");

            // require:
            Assert.AreEqual(0, cluster.GetSrcObjectCount());
            Assert.AreEqual(0, cluster.GetDstObjectCount());
            Assert.Null(cluster.GetLink());

            Assert.AreEqual(0, node.GetSrcObjectCount());
            Assert.AreEqual(0, node.GetDstObjectCount());

            // when:
            cluster.SetLink(node);

            // then:
            Assert.AreEqual(1, cluster.GetSrcObjectCount());
            Assert.AreSame(node, cluster.GetSrcObject(0));
            Assert.AreEqual(0, cluster.GetDstObjectCount());
            Assert.AreSame(node, cluster.GetLink());

            Assert.AreEqual(0, node.GetSrcObjectCount());
            Assert.AreEqual(1, node.GetDstObjectCount());
            Assert.AreSame(cluster, node.GetDstObject(0));
        }

        [Test]
        public void FbxCluster_Create_HasNamespacePrefix()
        {
            // given:
            var obj = new FbxCluster("asdf");

            // then:
            Assert.AreEqual("SubDeformer::", obj.GetNameSpacePrefix());
        }
    }
}
